### PR TITLE
Engageui 3797 topic prioritization

### PR DESCRIPTION
**Pull request by caitlanspronk**

### CODE REVIEW

**Comment by jensengar** [comment]
So this is a bit different than I had imagined, but I can see why it ended up this way. I think the SOP for creating a prioritized topic should happen when you actually subscribe, e.g. `subscribe: function (topic, handler, immediate = false, priority = 0) {...`. Is there a case for *changing* the priority of a topic after the initial subscription?

File: src/notifications.js [hunk 1]
```diff
@@ -279,6 +328,29 @@ export default class Notification {
...
+                oldPriorities[postfix] = newPriority;
+              }
+            });
+          } else {
+            const newTopics = {};
+            topicParts.postfixes.forEach(p => {
+              newTopics[p] = newPriority;
+            });
+            this.topicPriorities[topicParts.prefix] = newTopics;
+          }
```
Comment (jensengar) [inline src/notifications.js:352]: Do we need to resubscribe here? My thinking is if you added priority to a topic that wasn't high enough to make the cut list (more than 1k subscriptions), then I would think we would want to resubscribe to the new list of topics.

**Comment by caitlanspronk** [comment]
@jensengar I see what you're saying. I can add the priorities as parameter for both `subscribe` and `bulksubscribe` and not expose `setTopicPriorities`. I don't think we need to give consumers the ability to change a topic's priority after initial subscription.

File: src/notifications.js [hunk 2]
```diff
@@ -222,9 +253,48 @@ export default class Notification {
...
+        }, {});
+        this.topicPriorities[topicParts.prefix] = newTopics;
+      }
+    });
+  }
+
   get expose () {
     return {
-      subscribe: function (topic, handler, immediate) {
+      subscribe: function (topic, handler, immediate, priority) {
```
Comment (jensengar) [inline src/notifications.js:295]: will you add a test for this new param please?

**Pull request merged**